src/common/containers/builds.js

```diff
@@ -18,9 +18,9 @@ import styles from './container.css';
 class Builds extends Component {
   fetchInterval = null
 
-  fetchData({ snapLink, repository }) {
-    if (snapLink) {
-      this.props.dispatch(fetchBuilds(repository.url, snapLink));
+  fetchData({ snap, repository }) {
```

Comment:
Both `Builds` and `BuildDetails` share exactly the same logic for fetching snap and builds data.

Not sure if that's the right PR to do it, but it may be worth abstracting it.